profiles/tests/test_profiles_graphql_api.py

```diff
@@ -2317,6 +2318,163 @@ def test_staff_user_cannot_create_a_profile_with_sensitive_data_without_sensitiv
+                        }
+                    }
+                    phones {
+                        edges {
+                            node {
+                                phone
+                            }
+                        }
+                    }
+                    addresses {
+                        edges {
+                            node {
+                                address
+                            }
+                        }
+                    }
+                    youthProfile {
+                        schoolClass
+                    }
+                    sensitivedata {
+                        ssn
+                    }
+                }
+            }
+        }
+    """
+    )
+    query = t.substitute(
+        service_type=ServiceType.YOUTH_MEMBERSHIP.name,
+        id=to_global_id("ProfileNode", profile.pk),
+        first_name=data["first_name"],
+        phone_id=to_global_id("PhoneNode", phone.pk),
+        email=data["email"]["email"],
+        email_type=data["email"]["email_type"],
+        primary=str(data["email"]["primary"]).lower(),
+        phone=data["phone"],
+        address_id=to_global_id(type="AddressNode", id=address.pk),
+        school_class=data["school_class"],
+        ssn=data["ssn"],
+    )
+    expected_data = {
+        "updateProfile": {
+            "profile": {
+                "firstName": data["first_name"],
+                "emails": {"edges": [{"node": {"email": data["email"]["email"]}}]},
+                "phones": {"edges": [{"node": {"phone": data["phone"]}}]},
+                "addresses": {"edges": []},
+                "youthProfile": {"schoolClass": data["school_class"]},
+                "sensitivedata": {"ssn": data["ssn"]},
+            }
+        }
+    }
+    executed = user_gql_client.execute(query, context=request)
+    assert executed["data"] == expected_data
+
+
+def test_normal_user_cannot_update_a_profile_using_update_profile_mutation(
+    rf, user_gql_client
+):
+    profile = ProfileFactory(first_name="Joe")
+    ServiceFactory(service_type=ServiceType.YOUTH_MEMBERSHIP)
+    request = rf.post("/graphql")
+    request.user = user_gql_client.user
+
+    t = Template(
+        """
+        mutation {
+            updateProfile(
+                input: {
+                    serviceType: ${service_type},
+                    profile: {
+                        id: \"${id}\",
+                        firstName: \"${first_name}\",
+                    }
+                }
+            ) {
+                profile {
+                    firstName
+                }
+            }
+        }
+    """
+    )
+    query = t.substitute(
+        service_type=ServiceType.YOUTH_MEMBERSHIP.name,
+        id=to_global_id("ProfileNode", profile.pk),
+        first_name="John",
+    )
+    executed = user_gql_client.execute(query, context=request)
+    assert "errors" in executed
+    assert executed["errors"][0]["message"] == _(
```

Comment:
Probably not necessary, but one extra thing that could be asserted is that the original value in the profile is still there and was not changed.